src/main/java/com/google/sps/DetectTextFromImage.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.sps;
+
+import com.google.cloud.vision.v1.AnnotateImageRequest;
+import com.google.cloud.vision.v1.AnnotateImageResponse;
+import com.google.cloud.vision.v1.BatchAnnotateImagesResponse;
+import com.google.cloud.vision.v1.EntityAnnotation;
+import com.google.cloud.vision.v1.Feature;
+import com.google.cloud.vision.v1.Image;
+import com.google.cloud.vision.v1.ImageAnnotatorClient;
+import com.google.cloud.vision.v1.ImageSource;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class generates a shopping list based on image
+ * 1) It uses cloudVisionAPI to scan an image containing shopping list items and detect text from it. 
+ * 2) It then uses an algorithm to create shopping sentences (queries) from the text and their position.
+ * 3) This list of queries is returned to the Servlet from {@code imageToShoppingListExtractor()}.
+ */
+public class DetectTextFromImage {
+  private final String IMAGE_BASE_URI =
+      "https://shop-by-photos-step-2020.ey.r.appspot.com/get-image-url?blob-key=";
+  private final Feature TEXT_DETECTION_FEATURE = Feature.newBuilder().setType(Feature.Type.TEXT_DETECTION).build();
+  private ImageSource shoppingImageSource;
+
+  /*
+  * This constructor creates object only if a valid ImageKey is provided. 
+  * It then initializes shoppingImageSource to be used by the Generator function.
+  * If the ImageKey is invalid/mull, it throws an exception and the object never gets created.
+  */
+  public DetectTextFromImage(String shoppingImageKey) throws PhotoShoppingException {
+    if (!isValidImageKey(shoppingImageKey)) {
+      throw new PhotoShoppingException("ERROR : Invalid blob key");
+    }
+    shoppingImageSource = ImageSource.newBuilder().setImageUri(IMAGE_BASE_URI + shoppingImageKey).build();
```

Comment:
This will no longer be necessary after using image bytes to build the Image object. Personally, I use image bytes to build my image in the class (I have my implementation based on the current final workflow)

Reply:
I'll change my code to bytes after I finish all other review commits and get the basic code approved first.
What do you think @ChangchengXiao ?